addons/mrp/static/src/mrp_forecasted/forecasted_buttons.js

```diff
@@ -5,9 +5,15 @@ import { onWillStart } from "@odoo/owl";
 patch(ForecastedButtons.prototype, {
     setup() {
         super.setup();
-        onWillStart(async () =>{
+        onWillStart(async () => {
             const fields = this.resModel === "product.template" ? ['bom_ids'] : ['bom_ids', 'variant_bom_ids'];
-            const res = (await this.orm.call(this.resModel, 'read', [this.productId], { fields }))[0];
+            const productId =
+                (this.resModel !== "product.template" &&
+                 this.context.active_model === "product.template" &&
+                 this.context.variant_id)
+                    ? this.context.variant_id
+                    : this.productId;
```

Comment:
```suggestion
                const productId = this.resModel !== "product.template" &&
                    this.context.active_model === "product.template" &&
                    this.context.variant_id ? this.context.variant_id : this.productId;
```

More readable, and avoid unnecessary diff.